app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEnrollmentListFragment.kt

```diff
@@ -0,0 +1,121 @@
+import com.ustadmobile.core.impl.UMAndroidUtil
+import com.ustadmobile.core.view.ClazzEnrollmentListView
+import com.ustadmobile.lib.db.entities.ClazzEnrollment
+import com.ustadmobile.lib.db.entities.Person
+import com.ustadmobile.port.android.view.ext.navigateToEditEntity
+import com.ustadmobile.port.android.view.ext.setSelectedIfInList
+import com.ustadmobile.port.android.view.util.ListHeaderRecyclerViewAdapter
+import com.ustadmobile.port.android.view.util.SelectablePagedListAdapter
+
+
+class ClazzEnrollmentListFragment(): UstadListViewFragment<ClazzEnrollment, ClazzEnrollment>(),
+        ClazzEnrollmentListView, MessageIdSpinner.OnMessageIdOptionSelectedListener, View.OnClickListener{
+
+    private var mPresenter: ClazzEnrollmentListPresenter? = null
+
+    override val listPresenter: UstadListPresenter<*, in ClazzEnrollment>?
+        get() = mPresenter
+
+
+    class ClazzEnrollmentRecyclerAdapter(var presenter: ClazzEnrollmentListPresenter?): SelectablePagedListAdapter<ClazzEnrollment, ClazzEnrollmentRecyclerAdapter.ClazzEnrollmentListViewHolder>(DIFF_CALLBACK) {
+
+        class ClazzEnrollmentListViewHolder(val itemBinding: ItemClazzEnrollmentListBinding): RecyclerView.ViewHolder(itemBinding.root)
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ClazzEnrollmentListViewHolder {
+            val itemBinding = ItemClazzEnrollmentListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+            itemBinding.presenter = presenter
+            itemBinding.selectablePagedListAdapter = this
+            return ClazzEnrollmentListViewHolder(itemBinding)
+        }
+
+        override fun onBindViewHolder(holder: ClazzEnrollmentListViewHolder, position: Int) {
+            val item = getItem(position)
+            holder.itemBinding.clazzEnrollment = item
+            holder.itemView.setSelectedIfInList(item, selectedItems, DIFF_CALLBACK)
+        }
+
+        override fun onDetachedFromRecyclerView(recyclerView: RecyclerView) {
+            super.onDetachedFromRecyclerView(recyclerView)
+            presenter = null
+        }
+
+    }
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = super.onCreateView(inflater, container, savedInstanceState)
+
+        mPresenter = ClazzEnrollmentListPresenter(requireContext(), UMAndroidUtil.bundleToMap(arguments),
+                this, di, viewLifecycleOwner)
+
+        mDataRecyclerViewAdapter = ClazzEnrollmentRecyclerAdapter(mPresenter)
+
+        val createNewText = requireContext().getString(R.string.create_new,
+                requireContext().getString(R.string.clazzenrollment))
+
+        mUstadListHeaderRecyclerViewAdapter = ListHeaderRecyclerViewAdapter(
+                this, createNewText)
+        return view
+    }
+
+    override fun onResume() {
+        super.onResume()
+        mActivityWithFab?.activityFloatingActionButton?.text =
```

Comment:
Is this part of the template? This should be handled by the FabManager, which uses the fragment's lifecycle. This onResume event is the onResume event of the attached activity, not the fragment.